PhpUnit/ContainerBuilderHasFactoryConstraint.php

```diff
@@ -175,4 +176,30 @@ private function evaluateFactoryClass(ContainerBuilder $containerBuilder, $retur
         return true;
     }
 
+    private function getFactoryData( Definition $definition )
+    {
+        if( self::isLegacySymfonyDI() ) {
```

Comment:
This is a dirty, but working solution ... I checked with DI 2.3, 3.0 and 3.3 and they working ... a better approach may be an abstract class with "getFactoryData" as abstract method, but this commit resolves <= 2.6 versions and variated factory syntax in yml.

A test case not cover is a service with "setFactoryClass" definition ... This is pending.